tests/tapdance/examples.c

```diff
@@ -0,0 +1,197 @@
+/* Copyright 2022 Jouke Witteveen
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+#include "examples.h"
+
+// clang-format off
+
+// Example 1
+
+void dance_egg(qk_tap_dance_state_t *state, void *user_data) {
+    if (state->count >= 100) {
+        // SEND_STRING("Safety dance!");
```

Comment:
If you don't mind, I suggest we keep them in. In my opinion, the ideal would be to have all this code in the repository only once (i.e. test from the documentation `doctest` style). This is currently not easily possible not only because no tooling is in place, but also because the example code cannot be used for testing as-is. These commented out lines are a reminder of that.

Reply:
>  In my opinion, the ideal would be to have all this code in the repository only once (i.e. test from the documentation doctest style). This is currently not easily possible not only because no tooling is in place, but also because the example code cannot be used for testing as-is.

Agreed, having `doctest` ergonomics (which I know from Rust) would be a huge boon. Though I sadly don't see this happening in the short term.

Reply:
I added a line on top of `example.c` mentioning that the code is example code from the tap dance documentation.

Reply:
Thanks.